src/Engine/UnitActionLifecycles/BattleLifecycle.java

```diff
@@ -37,6 +37,33 @@ public static class BattleFactory extends UnitActionFactory
   {
     private static final long serialVersionUID = 1L;
 
+    // Provide an invalid FIRE option on selecting an action when:
+    // Weapon could attack, but nothing's in range
+    // Weapon is out of ammo
+    @Override
```

Comment:
Why bother showing a grayed-out menu item for units that can't fire? I'm no great fan of magically-disappearing menu items in general, but I've never felt that was a problem in the Advance Wars context; there really aren't alternate menus the mission option _might_ be in, and it's generally very clear why you can/can't do something.

Reply:
On current `master`, if I move onto an arty that's vaguely near the front and hit enter twice, one of two things happens:
I get the targeting reticle and can pick who I want to shoot.
I wait the arty in place.

With this change, the second possibility becomes "I am still on the action menu".
It's misclick prevention (and, for what it's worth, accurate to AW2)

Reply:
Okay, cool. makes sense.